app/code/local/Magmodules/Webwinkelconnect/Helper/Data.php

```diff
@@ -143,6 +143,50 @@ public function syncProductReview(array $product_review): array {
         }
     }
 
+
+    public function hasConsent(Mage_Sales_Model_Order $order): bool {
+        $permission_url = 'https://dashboard.webwinkelkeur.nl/api/2.0/order_permissions.json?' .
+            http_build_query([
+                'orderNumber' => $order->getIncrementId(),
+                'id' => $this->getShopId(),
+                'code' => $this->getApiKey(),
+            ]);
+
+        $start_time = microtime(true);
+        $ch = curl_init();
+        curl_setopt($ch, CURLOPT_TIMEOUT, 10);
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+        curl_setopt($ch, CURLOPT_FOLLOWLOCATION, true);
+        curl_setopt($ch, CURLOPT_FAILONERROR, true);
+        curl_setopt($ch, CURLOPT_URL, $permission_url);
```

Comment:
These calls to `curl_init` and `curl_setopt` can fail.  You should therefore check the return values:

~~~php
if (!$ch) {
  throw new RuntimeException("Could not initialize cURL");
}
~~~

~~~php
if (!curl_setopt($ch, CURLOPT_TIMEOUT, 10)) {
  throw new RuntimeException("Could not set cURL option");
}
~~~

To avoid repeating the check for `curl_setopt`, you can use `curl_setopt_array`:

~~~php
if (!curl_setopt_array($ch, [
  CURLOPT_TIMEOUT => 10,
  ...
])) {
  throw new RuntimeException(sprintf("Could not set cURL options: (%s) %s", curl_error($ch), curl_errno($ch)));
}
~~~